src/napari_spatialdata/_viewer.py

```diff
@@ -65,21 +67,34 @@ def inherit_metadata(self, layers: list[Layer]) -> None:
 
         show_info(f"Layer(s) without associated SpatialData object inherited SpatialData metadata of {ref_layer}")
 
-    def add_sdata_image(self, sdata: SpatialData, selected_cs: str, key: str) -> None:
-        affine = _get_transform(sdata.images[key], selected_cs)
-        rgb_image, rgb = _adjust_channels_order(element=sdata.images[key])
+    def add_sdata_image(self, sdata: SpatialData, selected_cs: str, key: str, multi: bool) -> None:
+        original_name = key
+        if multi:
+            original_name = original_name[: original_name.rfind("_")]
```

Comment:
same coment here regarding the validation. I think this code should be put in a function that does such validation.

Reply:
See above

Reply:
👍